module/VuFind/src/VuFind/RecordDriver/SolrMarc.php

```diff
@@ -72,4 +74,99 @@ public function getURLs()
             $this->getIlsURLs()
         );
     }
+
+    /**
+     * Takes a Marc field that notes are stored in (ex: 950) and a list of
+     * sub fields (ex: ['a','b']) optionally as well as what indicator
+     * number and value to filter for
+     * and concatonates the subfields together and returns the fields back
+     * as an array
+     * (ex: ['subA subB subC', 'field2SubA field2SubB'])
+     *
+     * @param string $field    Marc field to search within
+     * @param array  $subfield Sub-fields to return or empty for all
+     * @param string $indNum   The Marc indicator to filter for
+     * @param string $indValue The indicator value to check for
+     *
+     * @return array The values within the subfields under the field
+     */
+    public function getMarcFieldWithInd(
+        string $field,
+        ?array $subfield = null,
+        string $indNum = '',
+        string $indValue = ''
+    ) {
+        $vals = [];
+        $marc = $this->getMarcReader();
+        $marc_fields = $marc->getFields($field, $subfield);
+        foreach ($marc_fields as $marc_data) {
+            $field_vals = [];
+            if (trim(($marc_data['i' . $indNum] ?? '')) == $indValue) {
+                $subfields = $marc_data['subfields'];
+                foreach ($subfields as $subfield) {
+                    $field_vals[] = $subfield['data'];
+                }
+            }
+            if (!empty($field_vals)) {
+                $vals[] = implode(' ', $field_vals);
+            }
+        }
+        return array_unique($vals);
+    }
+
+    /**
+     * Get the abstract and summary notes
+     *
+     * @return array Note fields from the MARC record
+     */
+    public function getAbstractAndSummaryNotes()
+    {
+        return array_merge(
+            $this->getMarcFieldWithInd('520', null, '1', ''),
+            $this->getMarcFieldWithInd('520', null, '1', '0'),
+            $this->getMarcFieldWithInd('520', null, '1', '2'),
+            $this->getMarcFieldWithInd('520', null, '1', '3'),
+            $this->getMarcFieldWithInd('520', null, '1', '8'),
+        );
+    }
+
+    /**
+     * Get the location of other archival materials notes
+     *
+     * @return array Note fields from the MARC record
+     */
+    public function getLocationOfArchivalMaterialsNotes()
+    {
+        return array_merge(
+            $this->getMarcFieldWithInd('544', null, '1', ''),
+            $this->getMarcFieldWithInd('544', null, '1', '0')
+        );
+    }
+
+    /**
+     * Get the raw call numbers
+     *
+     * @return array Contents from the Solr field callnumber-raw
+     */
+    public function getCallNumbers()
```

Comment:
Funny, there is already a "`getCallNumbers()`" there which I had not noticed (with a similar implementation). Could we add a "`array_unique()`" to it to avoid duplicates ?

Reply:
Yes, I can't see a reason why that would be a problem!